src/runtime/mpz.cpp

```diff
@@ -30,6 +30,11 @@ mpz::mpz(int64 v) {
         mpz_neg(m_val, m_val);
 }
 
+size_t mpz::get_size_t() const {
+    static_assert(sizeof(size_t) <= sizeof(mp_limb_t), "GMP word size should be not greater than system word size");
+    return static_cast<size_t>(mpz_getlimbn(m_val, 0));
+}
```

Comment:
Your example does not seem to fulfill the `static_assert`. However, I see that I did screw up the ordering in the message.

Reply:
My point is that the conditions I listed satisfy `is_size_t`. 
It is not acceptable for `get_size_t` to fail if `is_size_t` is satisfied.

Reply:
But the function cannot fail in this case - the code would refuse to compile in the first place because of the `static_assert`. Yes, it would be nicer to have higher-level code that works on all imaginable platforms, but GMP only gives us `unsigned long` accessors. So either we add a messy ifdef for `sizeof(unsigned long) < sizeof(size_t)` platforms with some shifts generating suboptimal code, or we use this one direct access that CI tells us is safe on all platforms we're interested in. It is possible that the definition of `mp_limb_t` will change in the future, but I would argue the chance of this being incredibly low - and even then we will immediately know and can fix that line.

Reply:
> But the function cannot fail in this case - the code would refuse to compile in the first place because of the static_assert

Fail to compile is bad in this case since the current `static_assert` does not make sense. One would get the failure and then would find a mismatch between `is_size_t` and `get_size_t`. This must be fixed.

Reply:
> Yes, it would be nicer to have higher-level code that works on all imaginable platforms, but GMP only gives us unsigned long accessors. So either we add a messy ifdef for sizeof(unsigned long) < sizeof(size_t) platforms with some shifts generating suboptimal code

I am confused now. I thought you started to make these changes because on Windows `sizeof(unsigned long) < sizeof(size_t)`. So, as far as I can tell, this is already an issue. Or, is `sizeof(mp_limb_t) == sizeof(size_t)` on Windows?

Reply:
I'm not sure if you were referring to this, but I now see one potential in `is_size_t` on a hypothetical platform: if `sizeof(size_t) < sizeof(mp_limb_t)`, it would always return false for any positive value. So, here are the solutions I can see right now:
* I assume `sizeof(mp_limb_t) == sizeof(size_t)` in both functions and simplify `is_size_t` to `is_nonneg() && mpz_size(m_val) <= 1`.
* I completely avoid `mp_limb_t` and reimplement both functions with generic but ugly code.

Reply:
My problem is that the implementation of `is_size_t` does not match the `static_assert` at `get_size_t`.

> I assume sizeof(mp_limb_t) == sizeof(size_t) in both functions and simplify is_size_t to is_nonneg() && mpz_size(m_val) <= 1.

Yes, this is a reasonable solution, and we add a `static_assert` to both functions. 

BTW, does `static_cast<size_t>(mpz_getlimbn(m_val, 0))` behave correctly when `mpz_size(m_val) == 0`?